plugins/git/git.plugin.zsh

```diff
@@ -1,177 +1,188 @@
-compdef ggpnp=git
-
 # Pretty log messages
 function _git_log_prettily(){
   if ! [ -z $1 ]; then
     git log --pretty=$1
   fi
 }
-alias glp="_git_log_prettily"
-compdef _git glp=git-log
-
-# Work In Progress (wip)
-# These features allow to pause a branch development and switch to another one (wip)
-# When you want to go back to work, just unwip it
-#
 # This function return a warning if the current branch is a wip
 function work_in_progress() {
   if $(git log -n 1 2>/dev/null | grep -q -c "\-\-wip\-\-"); then
     echo "WIP!!"
   fi
 }
-# these alias commit and uncomit wip branches
-alias gwip='git add -A; git ls-files --deleted -z | xargs -r0 git rm; git commit -m "--wip--"'
-alias gunwip='git log -n 1 | grep -q -c "\-\-wip\-\-" && git reset HEAD~1'
 
-# these alias ignore changes to file
+#
+# Aliases
+# (sorted alphabetically)
+#
+
+alias g='git'
+
+alias ga='git add'
+
+alias gb='git branch'
+alias gba='git branch -a'
+alias gbl='git blame -b -w'
+alias gbnm='git branch --no-merged'
+alias gbr='git branch --remote'
+
+alias gc='git commit -v'
+alias gc!='git commit -v --amend'
+alias gca='git commit -v -a'
+alias gca!='git commit -v -a --amend'
+alias gcan!='git commit -v -a -s --no-edit --amend'
+alias gcl='git config --list'
+alias gclean='git reset --hard && git clean -dfx'
+alias gcm='git checkout master'
+alias gcmsg='git commit -m'
+alias gco='git checkout'
+alias gcount='git shortlog -sn'
+compdef gcount=git
+alias gcp='git cherry-pick'
+alias gcs='git commit -S'
+
+alias gd='git diff'
+alias gdc='git diff --cached'
+alias gdt='git diff-tree --no-commit-id --name-only -r'
+gdv() { git diff -w "$@" | view - }
+compdef _git gdv=git-diff
+alias gdw='git diff --word-diff'
+
+alias gf='git fetch'
+alias gfa='git fetch --all --prune'
+function gfg() { git ls-files | grep $@ }
+compdef gfg=grep
+
+alias gg='git gui citool'
+alias gga='git gui citool --amend'
+ggf() {
+[[ "$#" != 1 ]] && b="$(current_branch)"
+git push --force origin "${b:=$1}"
+}
+compdef _git ggf=git-checkout
+ggl() {
+[[ "$#" != 1 ]] && b="$(current_branch)"
+git pull origin "${b:=$1}"
+}
+compdef _git ggl=git-checkout
+ggp() {
+[[ "$#" != 1 ]] && b="$(current_branch)"
+git push origin "${b:=$1}"
+}
+compdef _git ggp=git-checkout
+ggpnp() {
+ggl "$1" && ggp "$1"
+}
+compdef _git ggpnp=git-checkout
+ggu() {
+[[ "$#" != 1 ]] && b="$(current_branch)"
+git pull --rebase origin "${b:=$1}"
+}
+compdef _git ggu=git-checkout
+
 alias gignore='git update-index --assume-unchanged'
-alias gunignore='git update-index --no-assume-unchanged'
-# list temporarily ignored files
 alias gignored='git ls-files -v | grep "^[[:lower:]]"'
+alias git-svn-dcommit-push='git svn dcommit && git push github master:svntrunk'
+compdef git-svn-dcommit-push=git
+
+alias gk='\gitk --all --branches'
+compdef _git gk='gitk'
+alias gke='\gitk --all $(git log -g --pretty=format:%h)'
+compdef _git gke='gitk'
+
+alias gl='git pull'
```

Comment:
For me this is confusing. gl is pull while glp is git log...
And you don't have gpr I use everyday (git pull --rebase)

I still think gp = git pull, gP = git push, gl = git log, is more logical.